glanzb/cards4-template.js

```diff
@@ -0,0 +1,142 @@
+var makeCard = // receive factory with external name `makeCard`
+    (function () { //begin IIFE...
+
+    // The factory itself:
+    function makeCard(id) {  //makeCard is also IIFE's internal name
+        if (!isValidID(id)) // abort if id is invalid
+            return null;
+        // set instance properties here
+        //...
+        // and return instance...
+        var card = {
+            id: id,
+            rank: rankFn,
+            suit: suitFn,
+            name: nameFn,
+            color: colorFn,
+            renderText: renderText, // renderText is the public name, fn is the internal
+            renderImage: renderImage
+        };
+    return card;
+    };
+
+    // makeCard(9)
+    // makeCard(9).rank()
+    // makeCard(9).name()
+//--------------------------
+// Private resources (internal use only)
+//--------------------------
+
+    // Examples:
+
+    //function isValidID(num) {...}
+    var isValidID = function(num) { // Returns--> true, false
+        return ((typeof num)==="number") //correct type
+                && (num%1 === 0)        //integer
+                && num>=0 && num<=51;   //in range
+    };
+
+    //var rankNames = [...];
+    var rankNames = ['','Ace','Two','Three','Four','Five','Six','Seven','Eight','Nine','Ten',
+                        'Jack','Queen','King'];
+    var suitNames = ['','Hearts','Diamonds','Spades','Clubs'];
+
+
+//-----------------------
+// Instance Methods: 
+//-----------------------
+
+//the functions just local variables within the iffe
+
+    function rankFn() {
+        return Math.floor(this.id/4)+1;
+    };
+    
+    function suitFn() {
+        return (this.id%4)+1;
+    };
+
+
+    function colorFn() { // -->"red,"black",NaN
+        var suit=this.suit();
+        return suit && ((suit<3)? "red": "black");
+    };
+
+    function nameFn() { //--> string, NaN
+        var rank = this.rank();
+        var suit = this.suit();
+        return rank && suit && (rankNames[rank]+' of '+ suitNames[suit]);
+    };
+
+    //etc...
+
+    //problem 4., mainly copied Dan's solution
+    var fileName = function(card) { //internal use only
+        var rankVal = card.rank(),
+            rankName = (rankVal>1 && rankVal<11);
```

Comment:
This `rankName` expression is where your problem with image names is coming from